hibernate-cache/src/test/java/org/infinispan/test/hibernate/cache/functional/AbstractFunctionalTest.java

```diff
@@ -239,4 +253,114 @@ protected CountDownLatch expectPutWithValue(AdvancedCache cache, Predicate<Objec
 			return new CountDownLatch(0);
 		}
 	}
+
+   protected CountDownLatch expectAfterEndInvalidation(AdvancedCache cache, int numInvalidates) {
+      CountDownLatch latch = new CountDownLatch(numInvalidates);
+      wrapInboundInvocationHandler(cache, handler -> new ExpectingInboundInvocationHandler(handler, latch));
```

Comment:
OK, I personally prefer the test to focus on test logic, instead of having a big `try { ... } finally { /* cleanup */ }` switch (which we have there anyway, though). Cleanup is what `@AfterMethod`s are for. But be it your way...